### ISSUE TITLE
IdenticalFalseToBooleanNotRector -> maybe break some code

### ISSUE THREAD

**Issue by voku** [rectorphp/rector#924]
"IdenticalFalseToBooleanNotRector" is enabled via level "coding-style" but ...

diff: e.g.
```diff
-        if (\strpos($str, "\xC2") === false) {
+        if (!\strpos($str, "\xC2")) {
             return $str;
         }
```

Here ```strpos("\xC2", "\xC2")``` will return 0 (first occurrence of the substring) and ! will catch it, but this breaks the code. 

PS: thanks for this awesome php tool, really good work :1st_place_medal:

**Comment by TomasVotruba** [comment]
Thanks for nice words, glad it's working from you!

I agree with you.
Maybe we could let PHPStan check for false|null static return values of function.

**Comment by TomasVotruba** [comment]
Fixed in #926
